src/gcp_scanner/crawl.py

```diff
@@ -368,23 +367,28 @@ def get_bucket_names(project_name: str, credentials: Credentials,
       break
 
     for bucket in response.get("items", []):
-      buckets_dict[bucket["name"]] = (bucket, None)
-      if dump_fd is not None:
-        ret_fields = "nextPageToken,items(name,size,contentType,timeCreated)"
```

Comment:
Why was this inside this if statement before and why is it outside now?

Reply:
in the original code, if a file descriptor is not provided, then the `ret_fields` and `req` variables are not needed, as the function does not need to retrieve and dump file information. But here in the modified code, the `ret_fields` and `req` variables are required to retrieve and store the root folder information for each bucket, regardless of whether a file descriptor has been provided. So this is moved outside here.

Reply:
Hm, I think there should be easier way to retrieve root folder of the given bucket. This way we generate much more traffic and noise

Reply:
you already have it under the bucket['name']

Reply:
The whole idea of dump_fd is to make user choose whether they want or not want to enable each file enumeration in the bucket to speed up the scanning. This change removes this ability. I suggest to keep it.

Reply:
@mshudrak   I got it. I have modified the code according to it. Please check